### PR TITLE
[DEV APPROVED] - Add the links for the countries component:

### PR DESCRIPTION
**Pull request by margOnline**
[TP 9517](https://moneyadviceservice.tpondemand.com/restui/board.aspx?#page=board/5624876525867731357&appConfig=eyJhY2lkIjoiNzEyRUZDMzlENEJERUE4QjJBRDVEQzdENUE1MEE2OEIifQ==&searchPopup=bug/9517)
### REQUIREMENT
Currently, the countries listed in the countries component do not link to anything. Each country needs to link to its regional page. 

Northern Ireland, Scotland and Wales link to regional_strategy pages; United Kingdon links to a uk_strategy page.

### SOLUTION
- in  the translation file, add the slug for each country
- amend the view to:
  - point to the slug defined in the translation file
  - use the correct layout path e.g. regional_strategy or uk_strategy
- re-record some cassettes as country pages were created in the cms for test purposes

### TECH DEBT
- refactor test files
  - move shared steps into a separate file
  - create a components directory
  - improve wording of some features

### NOTES FOR REVIEWING
- over half the changed files are either re-recorded vcrs or minor code tidying
- I considered several options for how to code the specific path for the country link and am undecided as to the best option. 
  1. add the path in the translations file
  1. add a method to the base_template to check if the country is 'uk' or not.
  1. use a helper method

I rejected option 2 because it requires the template to be passed every time the partial is invoked. Am undecided between options 1 and 3.

### CODE REVIEW

File: app/views/shared/_country_list.html.erb [hunk 1]
```diff
@@ -9,10 +9,24 @@
...
+                  uk_strategy_path(id: country[:slug], locale: 'en'),
+                  class: 'list--countries-link'
+              )%>
+              <% else %>
+                <%= link_to( 
+                    country[:text],
+                    regional_strategy_path(id: country[:slug], locale: 'en'),
+                    class: 'list--countries-link'
+                )%>
+            <% end %>
```
Comment (your) [inline app/views/shared/_country_list.html.erb]: The only difference between the two conditional branches is literally the path helper name – this made me thing that one way to DRY this is to have:

```ruby
<%= link_to(
    country[:text],
    strategy_path(country),
    class: 'list--countries-link'
)%>
```

where `strategy_path` could be an helper defined like:

```ruby
def strategy_path(country)
  if country[:region] == 'uk'
    uk_strategy_path(id: country[:slug], locale: 'en')
  else
    regional_strategy_path(id: country[:slug], locale: 'en')
  end
end
```

File: features/components/countries.feature [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  Scenario: Links - Regional Strategies
+    Given I entered a page which includes the countries component
+    When I click on "Northern Ireland" in the countries component
+    Then I should see the "Northern Ireland" page
+
+  Scenario: Links - United Kingdom
+    Given I entered a page which includes the countries component
+    When I click on "United Kingdom" in the countries component
+    Then I should see the "Uk Strategy" page
```
Comment (your) [inline features/components/countries.feature]: Love this test!

File: app/helpers/application_helper.rb [hunk 3]
```diff
@@ -13,4 +13,8 @@ def present(object, presenter_klass = nil)
   def present_collection(collection)
     collection.map { |element| present(element) }
   end
+
+  def region_uk?(string)
+    string == 'uk'
```
Comment (tomas-stefano) [inline app/helpers/application_helper.rb]: What do you think to rename this variable to region?

**Comment by margOnline** [comment]
@your @tomas-stefano ♻️

**Pull request merged**